test/e2e/e2e_test.go

```diff
@@ -697,6 +885,108 @@ var _ = Describe("e2e external gateway validation", func() {
+		}
+		framework.Logf("the pod side vtep node is %s and the ip %s", ciWorkerNodeSrc, localVtepIP)
+		podCIDR, err := getNodePodCIDR(ciWorkerNodeSrc)
+		if err != nil {
+			framework.Failf("Error retrieving the pod cidr from %s %v", ciWorkerNodeSrc, err)
+		}
+		framework.Logf("the pod cidr for node %s is %s", ciWorkerNodeSrc, podCIDR)
+		// setup the container to act as an external gateway and vtep
+		_, err = runCommand("docker", "exec", gwContainerName, "ip", "link", "add", "vxlan0", "type", "vxlan", "dev",
+			"eth0", "id", "4097", "dstport", vxlanPort, "remote", localVtepIP)
+		if err != nil {
+			framework.Failf("failed to create the vxlan interface on the test container: %v", err)
+		}
+		_, err = runCommand("docker", "exec", gwContainerName, "ip", "link", "set", "vxlan0", "up")
+		if err != nil {
+			framework.Failf("failed to enable the vxlan interface on the test container: %v", err)
+		}
+		_, err = runCommand("docker", "exec", gwContainerName, "ip", "address", "add", extGWCidr, "dev", "lo")
+		if err != nil {
+			framework.Failf("failed to add the external gateway ip to dev lo on the test container: %v", err)
+		}
+		_, err = runCommand("docker", "exec", gwContainerName, "ip", "route", "add", podCIDR, "dev", "vxlan0")
+		if err != nil {
+			framework.Failf("failed to add the pod route on the test container: %v", err)
+		}
+		// give the container time to come up and stabilize
+		time.Sleep(time.Second * 10)
+		By(fmt.Sprintf("Creating a container on %s and check default routes", ciWorkerNodeSrc))
+		framework.ExpectNoError(
+			// check pod has default routes going thru .3 address
+			checkHybridGWPodRoutes(f, ciWorkerNodeSrc, "hybrid-ext-gw-adds-routes-e2e", true))
+	})
+
+	It("Should not add default routes over .3 if both hybrid external gw and routing-external-gws annotations are set", func() {
+		// non-ha ci mode runs a set of kind nodes prefixed with ovn-worker
+		ciWorkerNodeSrc := ovnWorkerNode
+		if haMode {
+			// ha ci mode runs a named set of nodes with a prefix of ovn-control-plane
+			ciWorkerNodeSrc = ovnHaWorkerNode
+		}
+		localVtepIP, err := runCommand("docker", "inspect", "-f", ciNetworkFlag, ciWorkerNodeSrc)
+		if err != nil {
+			framework.Failf("failed to get the node ip address from node %s %v", ciWorkerNodeSrc, err)
+		}
+		localVtepIP = strings.TrimSuffix(localVtepIP, "\n")
+		if ip := net.ParseIP(localVtepIP); ip == nil {
+			framework.Failf("Unable to retrieve a valid address from container %s with inspect output of %s", gwContainerName, localVtepIP)
+		}
+		framework.Logf("the pod side vtep node is %s and the ip %s", ciWorkerNodeSrc, localVtepIP)
+		podCIDR, err := getNodePodCIDR(ciWorkerNodeSrc)
+		if err != nil {
+			framework.Failf("Error retrieving the pod cidr from %s %v", ciWorkerNodeSrc, err)
+		}
+		framework.Logf("the pod cidr for node %s is %s", ciWorkerNodeSrc, podCIDR)
+		// setup the container to act as an external gateway and vtep
```

Comment:
you dont really need to setup the docker container for this test right, you are just checking annotations.

Reply:
possibly, i copied over from previous test so must be it